### PR TITLE
Add people endpoints

### LINKED ISSUE #134: Expose `people` as entities
In order to expose public read token (#131), we have decided the easiest option will be to expose information about users.

Following data should be exposed:

```json
{
  "id": "username",
  "full_name": "Full Name",
  "public_token": "local access token"
}
```

Raionale:

- Clients like fittable would like to show some of the user's identity (e.g. full name), but forcing them to pull this from other source is overkill, since we already have these data.
- We can introduce a virtual resource e.g. `/people/me` which is mapped to the currently logged in user; clients then don't need to know authorised username at all but have an authoritative source.
- Other faculties don't work with Sirius directly but import data in batch. They may still want to show users their access tokens for iCal.

### PR DESCRIPTION
**Pull request by jnv**
This fixes #134. There are few notable changes:

- Required Postgres version is 9.4, due to dependency on pgcrypto method; this is okay for staging, but blocks deployment to production.
- Authorisation is _not properly tested_ since it is not possible: #137
- Access tokens are migrated to the users from the tokens table; this is because newly created users don't have access token under the current implementation and this is the most logical fix; however there is a big gap: we _expose_ tokens from people table, but _authorise_ using `tokens` table. I have attempted to fix this, but it causes failures in [totally unrelated code](https://travis-ci.org/cvut/sirius/jobs/80845158). I need time to investigate this.

### CODE REVIEW

File: db/migrate/1442325052_add_access_token_to_people.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    self << 'CREATE EXTENSION IF NOT EXISTS pgcrypto'
+    alter_table(:people) do
+      add_column :access_token, :uuid, default: Sequel.function(:gen_random_uuid), unique: true
+    end
+    self << <<-eos
+      UPDATE people
+      SET access_token = tokens.uuid
+      FROM tokens
+      WHERE people.id = tokens.username
+    eos
```
Comment (jirutka) [inline db/migrate/1442325052_add_access_token_to_people.rb]: Here-string is usually upper case, like EOS. :cat:

File: spec/api/people_endpoints_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+using Corefines::Enumerable::map_to
+
+describe API::PeopleEndpoints do
+  include_context 'API response'
+
+  subject { body }
+
+  describe 'GET /people' do
+
```
Comment (jirutka) [inline spec/api/people_endpoints_spec.rb]: There’s something missing, don’t you think? :smirk_cat:

File: spec/lib/sirius_api/people_authorizer_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  describe '#authorize_request!' do
+
+    context 'for READ_PERSONAL scope' do
+      it 'allows request to personal endpoint' do
+        expect { authorizer.authorize_request!(:get, '/people/:username', {username: 'ALICE'}) }.not_to raise_error
+      end
+
+      it 'denies request to endpoint for other user' do
+        expect { authorizer.authorize_request!(:get, '/people/:username', {username: 'BOB'}) }.to raise_error(SiriusApi::Errors::Authorization)
```
Comment (jirutka) [inline spec/lib/sirius_api/people_authorizer_spec.rb]: This line is too long (144 chars). Please keep it under 100 chars. You can improve it like this:

```rb
expect {
  authorizer.authorize_request!(:get, '/people/:username', {username: 'BOB'})
}.to raise_error(SiriusApi::Errors::Authorization)
```

File: spec/api/people_endpoints_spec.rb (hunk 2 above)
Comment (jnv) [inline spec/api/people_endpoints_spec.rb]: Yeah, I guess `pending` or perhaps I should remove it: https://github.com/cvut/sirius/pull/138/files#diff-abaa46834f1db586b22dda0666c459a8R18

**Comment by jirutka** [comment]
:shipit:

**Pull request merged**